src/wheel/bdist_wheel.py

```diff
@@ -432,38 +431,20 @@ def _ensure_relative(self, path):
 
     @property
     def license_paths(self):
-        metadata = self.distribution.get_option_dict("metadata")
+        metadata = self.distribution.metadata
         files = set()
-        patterns = sorted(
-            {option for option in metadata.get("license_files", ("", ""))[1].split()}
-        )
 
-        if "license_file" in metadata:
+        license_file = getattr(metadata, "license_file", None)
+        if license_file:
```

Comment:
Is this code path ever triggered on any version of setuptools?

Reply:
Yes, the user can specify `license_file` instead of `license_files` (it will already result in a warning from setuptools).

I can think about a unit test to add here... (I haven't done that before because the code path was basically inherited from the existing implementation).

Reply:
I'm asking because I looked and did not find this attribute on 45.2.0 or 57.0.0. So it would be nice to check if this gets triggered at all or if setuptools just lumps it into `license_files` on its own.

Reply:
We can see [here](https://github.com/pypa/setuptools/blob/d680efc8b4cd9aa388d07d3e298b870d26e9e04b/setuptools/dist.py#L642) that the `metadata` object has the attribute in the latest version (I am not sure about 45 or 57).

I will add some unit tests with the intention to activate this path, so we can have more certainty about it.

Reply:
BTW, we can see in the [same function](https://github.com/pypa/setuptools/blob/2451deee140b043dcb18796eabda370806881035/setuptools/dist.py#L583-L599) that setuptools is already adding `license_file` to `license_files`, so maybe `wheel` don't have to analyse it.

Reply:
In https://github.com/pypa/wheel/pull/466/commits/16826024e90fe08f3affd02766af807c5f978488, I added a check in the test suite to ensure the code path with the warning is triggered.

In https://github.com/pypa/wheel/pull/466/commits/5b09d963f3c678ffaa0602b7444e338792a4d86e, I removed that code path completely since it seems to already be handled by `setuptools` (the difference is the warning message exact wording and the warning class).

Reply:
> BTW, we can see in the [same function](https://github.com/pypa/setuptools/blob/v57.0.0/setuptools/dist.py#L583-L599) that setuptools is already adding `license_file` to `license_files`, so maybe `wheel` don't have to analyse it again...

I concur. I was just looking at the coverage report and I think the `license_files` property could be eliminated entirely.

Reply:
Perfect.

I removed this code path.

Please let me know if you prefer to remove entirely the `test_licenses_deprecated` (this way the `wheel` test suite is more independent of setuptools deprecation schedule). Tomorrow morning I can have a look at it.

Reply:
Yeah, we should not be testing setuptools code in the `wheel` test suite. Chuck it out :)

Reply:
Done in 16c90f4